tests/test_scanner.py

```diff
@@ -65,14 +64,48 @@ def test_process_output(self):
         output_thread.join.assert_called_once()
         error_thread.join.assert_called_once()
         process.wait.assert_called_once()
+        self.assertEqual(return_code, success_code)
 
-    def test_scan(self):
+    @patch("py_sonar_scanner.scanner.Thread")
+    def test_scan(self, mock_thread):
         scanner = Scanner(Configuration())
-
-        scanner.execute_command = Mock()
+        process = Mock()
+        scanner.execute_command = Mock(return_value=process)
         scanner.process_output = Mock()
 
         scanner.scan()
 
         scanner.execute_command.assert_called_once()
+        call_thread_stdout = call(target=scanner._log_output, args=(process.stdout,))
+        call_thread_stderr = call(target=scanner._log_output, args=(process.stderr,))
+        mock_thread.assert_has_calls([call_thread_stdout, call_thread_stderr])
+        self.assertEqual(mock_thread.call_count, 2)
         scanner.process_output.assert_called_once()
+
+    @patch("py_sonar_scanner.scanner.Popen")
+    def test_execute_command(self, mock_popen):
+        from subprocess import PIPE
+
+        scanner = Scanner(Configuration())
+        command = "test"
+        scanner.compute_command = Mock(return_value=command)
+
+        scanner.execute_command()
+
+        scanner.compute_command.assert_called_once()
+        mock_popen.assert_called_once_with(command, stdout=PIPE, stderr=PIPE)
```

Comment:
Perhaps here we could say the `stdout` and the `stderr` are both `py_sonar_scanner.scanner.PIPE` to be consistent with the patch annotation?

Reply:
Do you mean to have another patch annotation with`PIPE`? Because what I am trying to do in this test is check if the Popen constructor is called with the correct arguments, this is why I am using the real value of `PIPE`, but I don't want to call the real `Popen`, this is why I am patching it. 
When using patch, unittest wants us to provide the full name of the module using the patched object this is why we are patching "py_sonar_scanner.scanner.Popen". Tell me if this not super clear.

Reply:
Yes is understand, thank you. My initial confusion was regarding the reason for not using `subprocess.PIPE` in the patch annotation, but your comment clarifies this. 

Thank you!